### ISSUE TITLE
Early stop callback

### ISSUE THREAD

**Issue by CatStark** [jina-ai/finetuner#242]

**Comment by azayz** [comment]
Create ModelcheckpointCallback class which extends BaseCallback and does the following (or gives the possibility to choose from the following) : 

- [ ] save model (keras, pytorch and paddle) on every epoch end.
- [ ] save model (keras, pytorch and paddle) on every batch end. 
- [ ] choose whether to keep only best model.  
- [ ] choose whether to monitor train or validation
- [ ] choose whether to monitor other metrics different than loss
